src/server/TCGAadapter.js

```diff
@@ -125,22 +163,49 @@ export class TCGAAdapter {
   }
 
   correlations = (table, gene, min, max) => {
+    min = Number(min) || Number.MIN_SAFE_INTEGER
+    max = Number(max) || Number.MAX_SAFE_INTEGER
+    const conditions = {
+      gene: gene
+    }
+    const projection = {
+      rvalues: 1,
+      _id: 0
+    }
+    const query = [
+      { $match: conditions },
+      {
+        $project: {
+          rvalues: {
+            $filter: {
+              input: '$rvalues',
+              as: 'entry',
+              cond: {
+                $and: [
+                  { $ne: ['$$entry.rvalue', 'N/A'] },
+                  { $gt: ['$$entry.rvalue', min] },
+                  { $lt: ['$$entry.rvalue', max] }
+                ]
+              }
+            }
+          }
+        }
+      },
+      { $unwind: '$rvalues' },
+      { $sort: { 'rvalues.rvalue': -1 } },
+      { $group: { _id: '$_id', rvalues: { $push: '$rvalues' } } },
+      { $project: projection }
+    ]
     // Returns RBP names with corresponing Rvalue for gene in sorted order
-    return this.pool.query(`SELECT rvalue FROM ${table} WHERE gene = '${gene}'`).then(results => {
-      if (results.rows.length < 1) {
-        return [] // gene not found
-      }
-      // Pass just json with gene:correlation
-      const filteredRvals = rangeFilter(results.rows[0].rvalue, min, max)
-      // Map to object with gene and rvals as keys
-      const mappedRvals = _.map(filteredRvals, (value, gene) => {
-        return { gene: gene, Rvalue: value }
+    return this.mongodb
+      .collection(table)
+      .aggregate(query)
+      .toArray()
+      .then(results => {
```

Comment:
Right now the output of this query (`results`) looks something like:

```
[{"rvalues":[{"gene":"HNRNPAB","rvalue":0.468397770192776},{"gene":"SAMD14","rvalue":0.45895966910601},{"gene":"PARP12","rvalue":0.381557144877552},{"gene":"LSM4","rvalue":0.367733023644426}...
```
Can we just directly return the inner array, i.e.

```
[{"gene":"HNRNPAB","rvalue":0.468397770192776},{"gene":"SAMD14","rvalue":0.45895966910601}...
```

Reply:
I couldn't find any way to directly return the nested array

Reply:
Ok, I see, `aggregate` always returns an Object so this isn't possible inside mongo.